app/ptokens/views.py

```diff
@@ -29,39 +31,323 @@
+
+        if event_name == 'accept_redemption_ptoken':
+            if user.profile != redemption.ptoken.token_owner_profile:
+                return JsonResponse(
+                    {'error': _('You don\'t have permissions on the current redemption!')},
+                    status=401)
+            kwargs['redemption_accepted'] = datetime.now()
+            kwargs['redemption_state'] = 'accepted'
+            metadata['redemption'] = redemption.id
+        if event_name == 'denies_redemption_ptoken':
+            if user.profile != redemption.ptoken.token_owner_profile:
+                return JsonResponse(
+                    {'error': _('You don\'t have permissions on the current redemption!')},
+                    status=401)
+
+            kwargs['redemption_state'] = 'denied'
+            metadata['redemption'] = redemption.id
+        if event_name == 'complete_redemption_ptoken':
+            if user.profile != redemption.redemption_requester:
+                return JsonResponse(
+                    {'error': _('You don\'t have permissions on the current redemption!')},
+                    status=401)
+            web3_created = request.POST.get('web3_created')
+            if web3_created:
+                try:
+                    kwargs['web3_created'] = dateutil.parser.isoparse(web3_created)
+                except ValueError:
+                    return JsonResponse(
+                        {'error': _('Bad date format in web3_created')},
+                        status=401)
+            else:
+                kwargs['web3_created'] = datetime.now()
+
+            kwargs['redemption_state'] = 'completed'
+            kwargs['tx_status'] = request.POST.get('tx_status')
+            kwargs['txid'] = request.POST.get('txid')
+            metadata['redemption'] = redemption.id
+        elif event_name == 'tx_update':
+            kwargs['tx_status'] = request.POST.get('tx_status')
+
+        if kwargs:
+            RedemptionToken.objects.filter(pk=redemption.id).update(**kwargs)
+
+            if metadata:
+                record_ptoken_activity(event_name, redemption.ptoken, user.profile, metadata)
+
+    return JsonResponse({
+        'error': False,
+        'data': redemption.to_standard_dict()
+    })
+
+
+@csrf_exempt
+def ptoken_purchases(request, tokenId):
+    ptoken = get_object_or_404(PersonalToken, id=tokenId)
+    user = request.user if request.user.is_authenticated else None
+
+    if request.method == 'POST':
+        error = ''
+        if not user:
+            return JsonResponse(
+                {'error': _('You must be authenticated via github to use this feature!')},
+                status=401)
+
+        network = request.POST.get('network')
+        web3_created = request.POST.get('token_state')
+        token_holder_address = request.POST.get('token_holder_address')
+        txid = request.POST.get('txid')
+        tx_status = request.POST.get('tx_status')
```

Comment:
I think status must be static "pending" or None to be validated on background routine and filling this field according real transaction status, we need to check here if ptoken tx status is "success", or the status is checked by the contract?